eZ/Publish/Core/QueryType/QueryParameterContentViewQueryTypeMapper.php

```diff
@@ -61,13 +61,13 @@ private function extractParameters(ContentView $contentView)
 
     /**
      * @param ContentView $contentView
-     * @param string $queryParameterValue
+     * @param string|string[] $queryParameterValue
      *
      * @return mixed
      */
     private function evaluateExpression(ContentView $contentView, $queryParameterValue)
     {
-        if (substr($queryParameterValue, 0, 2) === '@=') {
+        if (is_string($queryParameterValue) && substr($queryParameterValue, 0, 2) === '@=') {
```

Comment:
Hi @clash82 I agree that the code breaks here but I won't do that this way because your code bypass the evaluateExpression for array.

As configuration allow to have multiple levels I guess array should be handled here.

Something like that should be more appropriate

```php
 /**
     * @param ContentView $contentView
     * @param string|array $queryParameterValue
     *
     * @return mixed
     */
    private function evaluateExpression(ContentView $contentView, $queryParameterValue)
    {
        if (is_array($queryParameterValue)) {
            $queryParameters = [];
            foreach ($queryParameterValue as $name => $value) {
                $queryParameters[$name] = $this->evaluateExpression($contentView, $value);
            }

            return $queryParameters;
        }

        if (substr($queryParameterValue, 0, 2) === '@=') {
            $language = new ExpressionLanguage();

            return $language->evaluate(
                substr($queryParameterValue, 2),
                [
                    'view' => $contentView,
                    'location' => $contentView->getLocation(),
                    'content' => $contentView->getContent(),
                ]
            );
        }
            
        return $queryParameterValue;
    }

```

/cc @bdunogier